libraries/bsg_manycore_platform.h

```diff
@@ -9,70 +10,89 @@
-                                 hb_mc_config_raw_t *config);
+        /**
+         * Receive a packet from manycore hardware
+         * @param[in] mc       A manycore instance initialized with hb_mc_manycore_init()
+         * @param[in] response A packet into which data should be read
+         * @param[in] timeout  A timeout counter. Unused - set to -1 to wait forever.
+         * @return HB_MC_SUCCESS on success. Otherwise an error code defined in bsg_manycore_errno.h.
+         */
+        int hb_mc_platform_receive(hb_mc_manycore_t *mc,
+                                   hb_mc_packet_t *packet,
+                                   hb_mc_fifo_rx_t type,
+                                   long timeout);
 
-/**
- * Stall until the all requests (and responses) have reached their destination.
- * @param[in]  mc     A manycore instance initialized with hb_mc_manycore_init()
- * @param[in] timeout A timeout counter. Unused - set to -1 to wait forever.
- * @return HB_MC_SUCCESS on success. Otherwise an error code defined in bsg_manycore_errno.h.
- */
-int hb_mc_platform_fence(hb_mc_manycore_t *mc, 
-                         long timeout);
+        /**
+         * Read the configuration register at an index
+         * @param[in]  mc     A manycore instance initialized with hb_mc_manycore_init()
+         * @param[in]  idx    Configuration register index to access
+         * @param[out] config Configuration value at index
+         * @return HB_MC_SUCCESS on success. Otherwise an error code defined in bsg_manycore_errno.h.
+         */
+        int hb_mc_platform_get_config_at(hb_mc_manycore_t *mc,
+                                         unsigned int idx,
+                                         hb_mc_config_raw_t *config);
+
+        /**
+         * Stall until the all requests (and responses) have reached their destination.
+         * @param[in]  mc     A manycore instance initialized with hb_mc_manycore_init()
+         * @param[in] timeout A timeout counter. Unused - set to -1 to wait forever.
+         * @return HB_MC_SUCCESS on success. Otherwise an error code defined in bsg_manycore_errno.h.
+         */
+        int hb_mc_platform_fence(hb_mc_manycore_t *mc,
+                                 long timeout);
+
+        /**
+         * Signal the hardware to start a bulk transfer over the network
+         * @param[in]  mc     A manycore instance initialized with hb_mc_manycore_init()
+         * @return HB_MC_SUCCESS on success. Otherwise an error code defined in bsg_manycore_errno.h.
+         */
+        int hb_mc_platform_start_bulk_transfer(hb_mc_manycore_t *mc);
+
+        /**
+         * Signal the hardware to end a bulk transfer over the network
+         * @param[in]  mc     A manycore instance initialized with hb_mc_manycore_init()
+         * @return HB_MC_SUCCESS on success. Otherwise an error code defined in bsg_manycore_errno.h.
+         */
+        int hb_mc_platform_finish_bulk_transfer(hb_mc_manycore_t *mc);
+}
 
```

Comment:
These are intended to abstract clock switching.

Reply:
nice